documentation/git_training.md

```diff
@@ -0,0 +1,258 @@
+
+1.  Check out to the branch that you stashed work from.
+
+2.  Once on the desired branch, you will use one of the commands below (depending on whether you named your stash or not).
+
+```{bash}
+#Retrive unnamed stash
+git stash pop
+
+#Retrive a named stash
+git stash apply stash^{/<stashName>}
+```
+
+These will return the work you had done previously to the branch. However, note these changes will not be available to other project users or on Git Hub which requires pushing (we will cover this momentarily).
+
+*Finding your current branch*
+
+If you are not sure what branch you are working on git branch will tell you what branch you are currently on. It's always a good idea to run git branch before pushing. The code below should highlight in green the branch you are currently on.
+
+```{bash}
+git branch
+```
+
+We created an issue in the economic mobility repository that we will use as a practice for the remaining part of this training. It is issue number 228. In your economic mobility folders lets git bash and create this branch.
+
+```{bash}
+git checkout -b iss228
+```
+
+## Updating and Pushing Your Work
+
+Now that we are on our practice branch, we are going to walk through the Git workflow.
+
+The most common Git commands you will use while working on updates are *git status, git add, git commit, and git push*. These four commands will be 95% of your git workflow. These tell Git which files to track, when to take snapshots of those files, and when to send those snapshots to GitHub for others to see.
+
+Once you have made updates to your code and it is ready for review you will go through the following steps. For the sake of practice, let's open a text file and save it to the main folder. Name it personalized to you "test_file\_\<your_name\>" and put whatever you want in it (maybe a little note saying how excited you are for this work).
+
+A helpful function git status prints out important information about your repo. Use it before and after most commands to understand how files have changed in your repo.
+
+```{bash}
+git status
+```
+
+We can see that the added text file is shown in red text. This tells us it is a change from how we cloned the repo and has not yet been prepared to be merged with our main repo.
+
+#### Adding and Committing
+
+To prepare our new file for the main repo we use git add which adds a file to Git's staging area. It tells Git, "Hey, look at this file." After adding the text file, if we check git status again the test_file.txt will now appear in green.
+
+```{bash}
+git add test_file.txt
+git status
+```
+
+Now that we have staged our file, we are ready to commit it. The command git commits commits changes made to added files to the repository. It says, "Hey, take a snapshot of the files I've added."
+
+```{bash}
+git commit -m "<message>"
+```
+
+Note: sometimes you may receive the message:
```

Comment:
Yay! Thanks for putting in these trouble-shooting items! I would like more of these because, although I do not recall seeing a message like this (probably because I've only been a reviewer), I have had other errors and didn't know how to get out of it.